src/download-sw.js

```diff
@@ -0,0 +1,82 @@
+/*
```

Comment:
Have you written this file on your own or is it taken from some other place?
In the first case please add here the same copyright notice as we have in `src/app.jsx` or so.
If not, then this might be a bit trickier.

Reply:
Haven't actually read this yet, just thought if it is necessary. We download archives for example in sosreport package - https://github.com/cockpit-project/cockpit/tree/master/pkg/sosreport

Reply:
> Have you written this file on your own or is it taken from some other place?
> In the first case please add here the same copyright notice as we have in `src/app.jsx` or so.
> If not, then this might be a bit trickier.

I've written this myself - I'll add the copyright notice.

Reply:
> Haven't actually read this yet, just thought if it is necessary. We download archives for example in sosreport package - https://github.com/cockpit-project/cockpit/tree/master/pkg/sosreport

That looks like it could be used to download the response, but I think it would start the download immediately after sending the request, causing the download to be stuck at 0 B/s until the checkpoint is done on the server- I'll try it.

Reply:
> I think it would start the download immediately after sending the request, causing the download to be stuck at 0 B/s until the checkpoint is done on the server- I'll try it.

I checked it and this doesn't happen, but there are other problems:

- The session timeouts when the checkpoint takes slightly longer (around 15 seconds). (I don't know whether this is expected - it might be a bug in the handling of external channels in cockpit-ws.)
- The Cockpit page doesn't get notified when the download starts (i.e. the checkpoint is finished).